### PR TITLE
Move parameters to ampdoc from viewer

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #22734, #22733.

### CODE REVIEW

File: src/inabox/amp-inabox.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ startupChunk(self.document, function initial() {
         installInaboxCidService(ampdoc);
         installViewerServiceForDoc(ampdoc);
         installInaboxViewportService(ampdoc);
-        installAmpdocServices(ampdoc, undefined, true);
+        installAmpdocServices(ampdoc, /* inabox */ true);
```
Comment (lannka) [inline src/inabox/amp-inabox.js]: seems you're on a pretty old master branch

File: src/service/ampdoc-impl.js [hunk 2]
```diff
@@ -47,15 +53,31 @@ export class AmpDocService {
...
-  constructor(win, isSingleDoc) {
+  constructor(win, isSingleDoc, opt_initParams) {
     /** @const {!Window} */
     this.win = win;
 
     /** @private {?AmpDoc} */
     this.singleDoc_ = null;
     if (isSingleDoc) {
-      this.singleDoc_ = new AmpDocSingle(win);
+      // Params can be passed via iframe hash/name with hash taking precedence.
```
Comment (lannka) [inline src/service/ampdoc-impl.js]: extract a method for readability?

File: test/unit/test-ampdoc.js [hunk 3]
```diff
@@ -54,17 +95,17 @@ describe('AmpDocService', () => {
...
-      expect(service.singleDoc_).to.be.instanceOf(AmpDocSingle);
+      expect(service.getSingleDoc()).to.exist;
+      expect(service.getSingleDoc()).to.be.instanceOf(AmpDocSingle);
     });
 
     it('should always yield the single document', () => {
       expect(() => service.getAmpDoc(null)).to.throw;
-      expect(service.getSingleDoc()).to.equal(service.singleDoc_);
-      expect(service.getAmpDoc(document)).to.equal(service.singleDoc_);
+      expect(service.getSingleDoc()).to.equal(service.getSingleDoc());
```
Comment (lannka) [inline test/unit/test-ampdoc.js]: seems can be removed? or are you testing it's the same instance? `to.be.same(xxx)`

File: test/unit/test-runtime.js [hunk 4]
```diff
@@ -1021,28 +1021,35 @@ describes.realWin(
...
-        ampdoc = new AmpDocShadow(win, docUrl, shadowRoot);
+        createShadowRoot(hostElement);
+        ampdoc = null;
 
         ampdocServiceMock
           .expects('installShadowDoc')
           .withExactArgs(
             docUrl,
-            sinon.match(arg => arg == getShadowRoot(hostElement))
+            sinon.match(arg => arg == getShadowRoot(hostElement)),
```
Comment (lannka) [inline test/unit/test-runtime.js]: change from `mock` to `spy` or `stub` might make the tests easier.

File: src/inabox/amp-inabox.js (hunk 1 above)
Comment (dvoytenko) [reply src/inabox/amp-inabox.js]: fixed

File: src/service/ampdoc-impl.js (hunk 2 above)
Comment (dvoytenko) [reply src/service/ampdoc-impl.js]: Good idea. Done.

File: test/unit/test-runtime.js (hunk 4 above)
Comment (dvoytenko) [reply test/unit/test-runtime.js]: done

File: test/unit/test-ampdoc.js (hunk 3 above)
Comment (dvoytenko) [reply test/unit/test-ampdoc.js]: You're right. A meaningless test.

**Pull request merged**